packages/notebook/src/browser/view-model/notebook-model.ts

```diff
@@ -314,7 +314,7 @@ export class NotebookModel implements Saveable, Disposable {
 
             // if selected cell is affected update it because it can potentially have been replaced
             if (cell === this.selectedCell) {
-                this.setSelectedCell(this.cells[cellIndex]);
+                this.setSelectedCell(this.cells.length > cellIndex ? this.cells[cellIndex] : this.cells[this.cells.length - 1]);
```

Comment:
Suggestion: Maybe something like this?
```suggestion
                this.setSelectedCell(this.cells[Math.min(Math.max(cellIndex, 0), this.cells.length - 1)]);
```

Reply:
why force it to be at least 0? accessing an array with index -1 just returns undefined which in this case is expected to happen